### PR TITLE
Add packages

### LINKED ISSUE #3657: Replace the default JSON Form error messages
## Is your feature request related to a problem? Please describe.

JSONForm is using the default AJV error messages.

## Describe the solution you'd like

The errors to be better than, for instance, when an integer is "a", the error message will read "should be integer"

## Implementation

TBd

## Describe alternatives you've considered

N/A

## Additional context

There's a few possibilities to investigate.

Want to keep in mind
- Ideally error messages can be customisable from the `uiSchema` or `jsonSchema` which will allow messages to be context driven when required
- Localisation is a factor
- As little work as possible

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #3657 

## Change summary

- Adds the ability to add a keyword into our JSON schema `errorMessage` like in [`ajv-errors`](https://www.npmjs.com/package/ajv-errors)

## Testing

Internal

### Related areas to think about

N/A

### CODE REVIEW

File: src/widgets/JSONForm/testJSON.js [hunk 1]
```diff
@@ -254,13 +257,15 @@ export const testTongaSurvey = {
           title: 'First Name',
           description: 'The first name of the respondent',
           type: 'string',
-          maxLength: 50,
+          maxLength: 2,
+          errorMessage: 'Hey! that names way to long, buddy!',
```
Comment (joshxg) [inline src/widgets/JSONForm/testJSON.js:261]: This is now the error message for this field. It's not specific to the error - which we can do if we want: https://www.npmjs.com/package/ajv-errors#messages-for-keywords

But I think for simplicity, just adding a single message might be OK?

File: src/widgets/JSONForm/JSONForm.js [hunk 2]
```diff
@@ -113,51 +127,97 @@ class FocusController {
...
+          formRef.current?.onSubmit(e);
+        },
+      }),
+      []
+    );
+
+    return (
+      <JSONFormContext.Provider value={options}>
+        <Form
+          validate={(formData, errorHandlers) => {
```
Comment (joshxg) [inline src/widgets/JSONForm/JSONForm.js:151]: This is all totally strange stuff to be able to add `ajv-errors` in .. :( Hopefully the comments help and not hinder

File: src/widgets/JSONForm/JSONForm.js [hunk 3]
```diff
@@ -14,6 +15,19 @@ import { PageButton } from '../PageButton';
...
+const ajvErrors = require('ajv-errors');
+
+const ajv = new Ajv({
+  errorDataPath: 'property',
+  allErrors: true,
+  multipleOfPrecision: 8,
+  schemaId: 'auto',
+  unknownFormats: 'ignore',
+  jsonPointers: true,
+});
```
Comment (joshxg) [inline src/widgets/JSONForm/JSONForm.js:27]: This is basically the same ajv instance which the json schema package uses

File: src/widgets/JSONForm/testJSON.js (hunk 1 above)
Comment (mark-prins) [inline src/widgets/JSONForm/testJSON.js:261]: :pedantic the 'to' should be 'too' in the too long and too short messages..

Comment (mark-prins) [reply src/widgets/JSONForm/testJSON.js:261]: single message is good

**Review by mark-prins — APPROVED** [review]
woo! it works. and is good.
Excited because I realised that the tester was screwing something up, I thought it wasn't working 🙄  and it was just poor testing.

👍

**Pull request merged**